File: backend/CookifyAPI/CookifyAPI/DTOs/Recipes/RecipeStepDto.cs
namespace CookifyAPI.DTOs.Recipes;

public class RecipeStepDto
{
    public int Id { get; set; }
    public string Title {get; set;}
    public int StepNumber { get; set; }
    public string Description { get; set; } = null!;
    public string? ImageUrl { get; set; }
}

File: frontend/lib/feature/recipe/recipe_common/data/models/recipe_step_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'recipe_step_dto.freezed.dart';
part 'recipe_step_dto.g.dart';

@freezed
abstract class RecipeStepDto with _$RecipeStepDto {
  const factory RecipeStepDto({
    required int id,
    required String? title,
    required String description,
    @JsonKey(name: 'image_url') required String imageUrl,
  }) = _RecipeStepDto;

  factory RecipeStepDto.fromJson(Map<String, dynamic> json) =>
      _$RecipeStepDtoFromJson(json);
}
